# Laboratorio 1 - AlpesHearth

## Integrantes

- Isabella Naranjo
- David Caro

## Exploración de los datos

Durante la fase de exploración de datos encontramos las siguientes irregularidades:
1. Muchos registros tienen la edad, el peso, la altura y el índice de masa corporal **(BMI)** vacios.
2. Las categorias booleanas vienen con strings como valor, es decir, "Y/N".
3.  

Ahora bien, para realizar una exploración más profunda utilizamos la librería pandas, corrimos los siguientes comandos:

In [5]:
import pandas as pd 
import numpy as np 

df = pd.read_csv("../data/Datos Lab 1.csv")
df.head()
df.shape
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1639 entries, 0 to 1638
Data columns (total 24 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Patient ID                    1639 non-null   object 
 1   Date of Service               1639 non-null   object 
 2   Sex                           1639 non-null   object 
 3   Age                           1571 non-null   float64
 4   Weight (kg)                   1566 non-null   float64
 5   Height (m)                    1578 non-null   float64
 6   BMI                           1586 non-null   float64
 7   Abdominal Circumference (cm)  1578 non-null   float64
 8   Blood Pressure (mmHg)         1639 non-null   object 
 9   Total Cholesterol (mg/dL)     1571 non-null   float64
 10  HDL (mg/dL)                   1557 non-null   float64
 11  Fasting Blood Sugar (mg/dL)   1585 non-null   float64
 12  Smoking Status                1639 non-null   object 
 13  Dia

De estas primeras líneas de código pudimos identificar lo siguiente:

1. Tenemos 1639 registros y 24 columnas (14 númericas y 10 categoricas).
2. ALgunas columnas tienen valores faltantes, podemos observar que hay valores nulos en la mayoría de columnas clínicas, es decir, edad, peso, altura etc...
3. Inclusive, hay 29 valores nulos en nuestra variable objetivo **CVD Risk Score**.
4. La variable **Date of Service** aparece como tipo object, sin embargo, fue documentada como Date y es necesario hacer un casting para poder manipularla.
5. Las variables **Systolic BP** y **Diastolic BP** son las componentes de la variable **Blood Pressure**, así que puede que podamos descarta dicha columna. 
6. La variable **CVD Risk Level** es categórica y probablemente sea derivada del **CVD Risk Score**, por lo que no será utilizada como variable predictora para evitar fuga de información.


Seguido a esto, corrimos las siguientes líneas de código:

In [19]:
print("Duplicados: ", df.duplicated().sum())
print("Valores presión sanguínea: ", df["Blood Pressure (mmHg)"].unique()[:10])
print("Valores máximos y mínimos de las variables clínicas:")
print("Edad: ", df["Age"].min(), ",", df["Age"].max())
print("Peso: ", df["Weight (kg)"].min(), ",", df["Weight (kg)"].max())
print("Altura: ", df["Height (m)"].min(), ",", df["Height (m)"].max())
print("Presión sistólica: ",df["Systolic BP"].min(), ",", df["Systolic BP"].max())
print("Presión diastólica: ", df["Diastolic BP"].min(), ",", df["Diastolic BP"].max())
print("Porcentaje de valores faltantes: ")
missing_percent = df.isnull().mean()*100
print(missing_percent.sort_values(ascending=False))

Duplicados:  151
Valores presión sanguínea:  ['112/83' '101/91' '90/74' '92/89' '121/68' '107/61' '139/81' '106/115'
 '103/99' '165/99']
Valores máximos y mínimos de las variables clínicas:
Edad:  6.134 , 89.42
Peso:  13.261 , 158.523
Altura:  1.371 , 2.146
Presión sistólica:  49.914 , 202.711
Presión diastólica:  31.72 , 134.066
Porcentaje de valores faltantes: 
Diastolic BP                    5.186089
HDL (mg/dL)                     5.003051
Waist-to-Height Ratio           4.636974
Weight (kg)                     4.453935
Height (cm)                     4.148871
Age                             4.148871
Total Cholesterol (mg/dL)       4.148871
Height (m)                      3.721782
Abdominal Circumference (cm)    3.721782
Systolic BP                     3.721782
Estimated LDL (mg/dL)           3.477730
Fasting Blood Sugar (mg/dL)     3.294692
BMI                             3.233679
CVD Risk Score                  1.769372
Patient ID                      0.000000
Blood Pressure Cate

Sobre estas nuevas línead de comandos, determinamos:
1. Hay 151 registros completamente duplicados, lo que representa aproximadamente el 9% del dataset, dichos registros deben eliminados para evitar sesgos en el entrenamiento del modelo.
2. Como lo mencionamos anteriormente, la variable **Blood Pressure** es un string que combina las variables **Systolic BP** y **Diastolic BP**, así que esta variable es redundante y podríamos eliminarla.
3. Se observan valores decimales en la variable **edad** (6.13 años), cosa que podría indicar errores de registro o transformaciones previas en los datos.
4. El valor mínimo de la variable peso es de 13.261kg, cosa que es sospechosa en vista de que estamos hablando de adultos.
5. El valor máximos de la presión sistólica fue de 202.711, dicho valor podría ser un outlier.
6. Hay valores nulos en múltiples variables clínicas, cosa que requiere estrategias de imputación en la fase de preparación de datos, adicionalmente, la variable objetivo CVD Risk Score presenta aproximadamente 1.8% de valores nulos, dichos valores deberan ser tratados con cuidado.